packages/flutter_tools/lib/src/cache.dart

```diff
@@ -1598,10 +1598,11 @@ class ArtifactUpdater {
       );
       try {
         _ensureExists(tempFile.parent);
-        final IOSink ioSink = tempFile.openWrite();
-        await _download(url, ioSink);
-        await ioSink.flush();
-        await ioSink.close();
+        if (tempFile.existsSync()) {
+          tempFile.deleteSync();
```

Comment:
Needs to be deleted if it exists so that the downloader does not append to a file that already exists